src/python/WMCore/Database/CMSCouch.py

```diff
@@ -1117,82 +1128,90 @@ def __init__(self, reason, data, result, status=None):
+        errorMsg += f"Error type: {self.type}, Status code: {self.status}, "
+        errorMsg += f"Reason: {self.reason}, Data: {repr(self.data)}"
+        return errorMsg
 
 
 class CouchBadRequestError(CouchError):
-    def __init__(self, reason, data, result):
-        CouchError.__init__(self, reason, data, result)
+    def __init__(self, reason, data, result, status):
+        CouchError.__init__(self, reason, data, result, status)
         self.type = "CouchBadRequestError"
 
 
 class CouchUnauthorisedError(CouchError):
-    def __init__(self, reason, data, result):
-        CouchError.__init__(self, reason, data, result)
+    def __init__(self, reason, data, result, status):
+        CouchError.__init__(self, reason, data, result, status)
         self.type = "CouchUnauthorisedError"
 
 
 class CouchNotFoundError(CouchError):
-    def __init__(self, reason, data, result):
-        CouchError.__init__(self, reason, data, result)
+    def __init__(self, reason, data, result, status):
+        CouchError.__init__(self, reason, data, result, status)
         self.type = "CouchNotFoundError"
 
 
 class CouchNotAllowedError(CouchError):
-    def __init__(self, reason, data, result):
-        CouchError.__init__(self, reason, data, result)
+    def __init__(self, reason, data, result, status):
+        CouchError.__init__(self, reason, data, result, status)
         self.type = "CouchNotAllowedError"
 
 class CouchNotAcceptableError(CouchError):
-    def __init__(self, reason, data, result):
-        CouchError.__init__(self, reason, data, result)
+    def __init__(self, reason, data, result, status):
+        CouchError.__init__(self, reason, data, result, status)
         self.type = "CouchNotAcceptableError"
 
 class CouchConflictError(CouchError):
-    def __init__(self, reason, data, result):
-        CouchError.__init__(self, reason, data, result)
+    def __init__(self, reason, data, result, status):
+        CouchError.__init__(self, reason, data, result, status)
         self.type = "CouchConflictError"
 
 
 class CouchFeatureGone(CouchError):
-    def __init__(self, reason, data, result):
-        CouchError.__init__(self, reason, data, result)
+    def __init__(self, reason, data, result, status):
+        CouchError.__init__(self, reason, data, result, status)
         self.type = "CouchFeatureGone"
 
 
 class CouchPreconditionFailedError(CouchError):
-    def __init__(self, reason, data, result):
-        CouchError.__init__(self, reason, data, result)
+    def __init__(self, reason, data, result, status):
+        CouchError.__init__(self, reason, data, result, status)
         self.type = "CouchPreconditionFailedError"
 
+
+class CouchRequestTooLargeError(CouchError):
+    def __init__(self, reason, data, result, status):
+        # calculate the size of this JSON serialized object
+        docSize = sys.getsizeof(json.dumps(data))
+        errorMsg = f"Document has {docSize} bytes and it's too large to be accepted by CouchDB."
```

Comment:
here it is insufficient to print document has certain size since you need to specify CouchDB threshold otherwise you can't clearly define what is too large means.

Reply:
CouchDB does not tell me what the configured limit is, so I have no way to say that in the error message either. The best I can do is to say something "CouchDB limit is defined by couchdb.max_document_size setting (default 8M bytes)".